modules/core/02-client/keeper/grpc_query.go

```diff
@@ -63,25 +63,26 @@ func (q Keeper) ClientStates(c context.Context, req *types.QueryClientStatesRequ
 	clientStates := types.IdentifiedClientStates{}
 	store := prefix.NewStore(ctx.KVStore(q.storeKey), host.KeyClientStorePrefix)
 
-	pageRes, err := query.Paginate(store, req.Pagination, func(key, value []byte) error {
+	pageRes, err := query.FilteredPaginate(store, req.Pagination, func(key, value []byte, accumulate bool) (bool, error) {
```

Comment:
Would it be possible to modify [the existing test](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/grpc_query_test.go#L131-L154) to check for the `total` value in the `pagination` field of the response (i.e. `total` doesn't account for `consensusStates`)?

Maybe could also add another test with two clients that sets `limit` to 1 and checks that the `nextKey` in `pagination` points to a key with the format `/<client-id>/clientState`. But I understand that maybe this is a bit too much, and anyway maybe it's testing functionality from `Paginate`, not really our gRPC endpoint, so we can also skip it.